### PR TITLE
fix(TPSVC-7920/CollapsiblePanel): use onEntered/onExited

### PR DESCRIPTION
**Pull request by lmaillet**
**What is the problem this PR is trying to solve?**
Collapsible panel in VirtualizedList does not work in Firefox and Edge
See https://jira.talendforge.org/browse/TPSVC-7920
Because it use setTimeout with a cross fingers renderer has happened.

**What is the chosen solution to this problem?**

Force the VirtualizedList to measure again the panel after each expand or collapse using `onEntered` or  `onExited` callback (https://react-bootstrap.github.io/utilities/transitions/#collapse).

There is still a problem in storybook if you load directly the page, because storybook loads the css after javascript and so css is loaded after the first render. But it works in _normal_ app.

@jmfrancois I've removed the `appear` prop, as it breaks the UT and looks like it's not needed anymore.
image

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 1]
```diff
@@ -62,7 +58,12 @@ class RowCollapsiblePanel extends React.Component {
 						aria-label={get(rowData, 'header[0].label')}
 						style={style}
 					>
-						<PureCollapsiblePanel {...{ rowData, measure, onToggle: this.onToggle }} />
+						<CollapsiblePanel
+							onEntered={() => onMeasure(measure)}
+							onExited={() => onMeasure(measure)}
```
Comment (jmfrancois) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: could you please try onEntererd={measure} / same for onExited and remove onMeasure ?

Comment (lmaillet) [reply packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: Done 👍

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 2]
```diff
@@ -62,7 +55,12 @@ class RowCollapsiblePanel extends React.Component {
 						aria-label={get(rowData, 'header[0].label')}
 						style={style}
 					>
-						<PureCollapsiblePanel {...{ rowData, measure, onToggle: this.onToggle }} />
+						<CollapsiblePanel
+							onEntered={measure}
```
Comment (sgendre) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: I rather name it onEnter

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js [hunk 3]
```diff
@@ -62,7 +55,12 @@ class RowCollapsiblePanel extends React.Component {
...
+							onExited={measure}
```
Comment (sgendre) [inline packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: I rather name it onExit

Comment (lmaillet) [reply packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: I used `onEntered` and `onExited` to keep consistency with bootstrap collapse.
`onEnter` callback is called **before** component expands and `onEntered` is called **after** component has expanded. Same for `onExit` and `onExited`.
And I would like `react-virtualized` to measure the row after expand/collapse.
See https://react-bootstrap.github.io/utilities/transitions/#collapse

File: packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js (hunk 2 above)
Comment (jsomsanith) [reply packages/components/src/VirtualizedList/RowCollapsiblePanel/RowCollapsiblePanel.component.js]: It seems that `react-bootstrap` collapsible panel that our component use, has a prop `onEntered`. Let's not change the name to do a mapping in our component.

**Pull request merged**